webknossos/tests/test_examples.py

```diff
@@ -21,38 +23,45 @@ def tmp_cwd() -> Iterator[None]:
             os.chdir(prev_cwd)
 
 
+def exec_module_as_main(module: ModuleType) -> None:
+    source = inspect.getsource(module)
+    new_source = source.replace('\nif __name__ == "__main__":\n', "\nif True:\n")
+    exec(new_source, module.__dict__)  # pylint: disable=exec-used
```

Comment:
alternatively, you could define a `def main():` in the examples which can be imported then. I would find this more elegant, but I can understand if you want to stick to the current approach for the sake of keeping the examples more "script-like".

Reply:
I actually had that in place first, but got problem with testing: To assert the variables one needs them to be returned or in the global scope, which I both found weird in the example. However, I could do a similar trick with changing the code of the module to do any of those options, I'll give it another try.

Reply:
I think, returning something in the examples would be fine. The example could even use it like this:

```
def main():
  ...
  return some_object

if main...:
  some_object = main()
  print(some_object)
```

Reply:
I changed this now to use the `main` function, but without adding returns in the examples (I think they might just confuse users). I used the same magic to make all needed variables global in the tests. Adding a return statement works, but the replace-code is much more fragile, since the end of the function needs to be detected.